tests/pimple.spec.ts

```diff
@@ -45,40 +63,58 @@ describe('pimple container', () => {
     });
 
     it('returns true if service exists', async () => {
-        const container = new Pimple();
+        type ServiceMap = {
+            'foo': number,
+        };
+
+        const container = new Pimple<ServiceMap>();
 
         expect(container.has('foo')).toBe(false);
     });
 
     it('protects function', async () => {
-        const container = new Pimple();
-
-        const protectedFunciton = () => {
-            return 'baz';
+        type ServiceMap = {
+            foo: () => string,
+            untyped: Function,
         };
 
-        container.set('foo', container.protect(protectedFunciton));
+        const container = new Pimple<ServiceMap>();
+
+        const typeProtectedFunction = () => 'baz';
+        const untypedProtectedFunction = () => 42;
+
+        container.set('foo', container.protect('foo', typeProtectedFunction));
+        container.set('untyped', container.protect('untyped', untypedProtectedFunction));
```

Comment:
This should be ideally:

```suggestion
        container.set('foo', container.protect(typeProtectedFunction));
        container.set('untyped', container.protect(untypedProtectedFunction));
```